VariantShredding.md

```diff
@@ -25,290 +25,318 @@
-There is one metadata value for the entire Variant record, and that is stored in the top-level `metadata` field.
-This means any `variant_value` values in the shredded representation is only the "value" portion of the [Variant Binary Encoding](VariantEncoding.md).
+The series of `tags` arrays `["comedy", "drama"], ["horror", null], ["comedy", "drama", "romance"], null` would be stored as:
 
-The metadata is kept at the top-level, instead of shredding the metadata with the shredded variant values because:
-* Simplified shredding scheme and specification. No need for additional struct-of-binary values, or custom concatenated binary scheme for `variant_value`.
-* Simplified and good performance for write shredding. No need to rebuild the metadata, or re-encode IDs for `variant_value`.
-* Simplified and good performance for Variant reconstruction. No need to re-encode IDs for `variant_value`.
+| Array                            | `value`     | `typed_value `| `typed_value...value` | `typed_value...typed_value`    |
+|----------------------------------|-------------|---------------|-----------------------|--------------------------------|
+| `["comedy", "drama"]`            | null        | non-null      | [null, null]          | [`comedy`, `drama`]            |
+| `["horror", null]`               | null        | non-null      | [null, `00`]          | [`horror`, null]               |
+| `["comedy", "drama", "romance"]` | null        | non-null      | [null, null, null]    | [`comedy`, `drama`, `romance`] |
+| null                             | `00` (null) | null          |                       |                                |
 
-# Data Skipping
+#### Objects
 
-Shredded columns are expected to store statistics in the same format as a normal Parquet column.
-In general, the engine can only skip a row group or page if all rows in the `variant_value` field are null, since it is possible for a `variant_get` expression to successfully cast a value from the `variant_value` to the target type.
-For example, if `typed_value` is of type `int64`, then the string “123” might be contained in `variant_value`, which would not be reflected in statistics, but could be retained by a filter like `where variant_get(col, “$.field”, “long”) = 123`.
-If `variant_value` is all-null, then the engine can prune pages or row groups based on `typed_value`.
-This specification is not strict about what values may be stored in `variant_value` rather than `typed_value`, so it is not safe to skip rows based on `typed_value` unless the corresponding `variant_value` column is all-null, or the engine has specific knowledge of the behavior of the writer that produced the shredded data.
+Fields of an object can be shredded using a Parquet group for `typed_value` that contains shredded fields.
 
-# Shredding Semantics
+If the value is an object, `typed_value` must be non-null.
+If the value is not an object, `typed_value` must be null.
```

Comment:
```suggestion
If the value is not an object, `typed_value` must be null [1].

[1] These requirements allows readers to easily distinguish between empty objects (`{}`) and other variant types when attempting to project an object and all the projected object fields have been shredded.
```
Not sure if this is valid markdown but I think it is worth capturing context. But I think this captures the reasoning explained below (phrasing on footnote can probably be improved)?

Reply:
I moved the line that is now just below this to expand on it.